examples/serde_compat.rs

```diff
@@ -0,0 +1,125 @@
+//! An example of implementing `serde::Serialize` and `serde::Deserialize` equivalently to how
+//! `#[derive(Serialize, Deserialize)]` would on `bitflags` `1.x` types.
+
+#[cfg(feature = "serde")]
+fn main() {
+    bitflags::bitflags! {
+        // Removed: `serde` traits from the `#[derive]` attribute
+        // #[derive(Serialize, Deserialize)]
+        #[derive(Debug, PartialEq, Eq)]
+        pub struct Flags: u32 {
+            const A = 1;
+            const B = 2;
+            const C = 4;
+            const D = 8;
+        }
+    }
+
+    // Added: Manual `Serialize` and `Deserialize` implementations based on a generic impl
+    impl serde::Serialize for Flags {
+        fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            legacy_format::serialize(self, serializer)
+        }
+    }
+
+    impl<'de> serde::Deserialize<'de> for Flags {
+        fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+            legacy_format::deserialize(deserializer)
+        }
+    }
+
+    pub mod legacy_format {
```

Comment:
I think I will pull this out into a library to make it easier to consume. Users would then be free to inline it from there if they wanted to.